firestore.rules

```diff
@@ -1,13 +1,53 @@
 rules_version = '2';
 service cloud.firestore {
   match /databases/{database}/documents {
+    function validAmount(docData){
+      return docData.amount is map 
+      && docData.amount.type in ['UNKNOWN', 'FIXED','VARIES','FULL_TUITION']
+      && docData.amount.min is int
+      && docData.amount.max is int
+    }
+
+    function validReqs(docData){
+      return !('requirements' in docData) 
+      || docData.requirements is map
+      && (!('gpa' in docData.requirements) || docData.requirements.gpa is number && docData.requirements.gpa > 0 && docData.requirements.gpa <= 4)
+      && (!('ethnicities' in docData.requirements) || docData.requirements.ethnicities is list)
+      && (!('majors' in docData.requirements) || docData.requirements.majors is list)
```

Comment:
I wonder if you can refactor these "not set or is list" checks but that probably requires being able to do something like `reqs[someKey]`. However if you take in `docData.requirements` as simply `reqs` then maybe these lines won't look too long.